pattern/pattern.c

```diff
@@ -180,6 +256,21 @@ void pattern_render(struct pattern * pattern, GLuint input_tex) {
         loc = glGetUniformLocationARB(pattern->shader[i], "iChannel");
         glUniform1ivARB(loc, pattern->n_shaders, pattern->uni_tex);
 
+        if (pattern->frames) {
+            int texture_index = pattern->n_shaders + 1;
+            glActiveTexture(GL_TEXTURE0 + texture_index);
+            glBindTexture(GL_TEXTURE_2D, pattern->frames[pattern->current_frame]);
+
+            loc = glGetUniformLocationARB(pattern->shader[i], "iImage");
+            glUniform1iARB(loc, texture_index);
+
+            // TODO: Should do something interesting off of beat_frac and beat_index
+            // to make the animation go faster/slower depending on the music
+            if (pattern->last_ms / RADIANCE_PATTERN_GIF_SPEED != time_master.wall_ms / RADIANCE_PATTERN_GIF_SPEED) {
```

Comment:
I don't think that interpretation is correct?  In initialization we set `pattern->current_frame` to 0.  Then the next time we cross an interval boundary (of time `1/RADIANCE_PATTERN_GIF_SPEED` ms if my math is right) we increment the current frame by 1.  So I don't think we jump, although we do spend a smaller interval on the first frame rendered (which is fine).  I've confirmed this statement with a `printf("%d\n", pattern->current_frame);` right before the `glBindTexture`.